lib/public_urls.rb

```diff
@@ -1,26 +1,28 @@
 module PublicUrls
   class << self
-    def url_for(base_path:)
-      URI.join(website_root, base_path).to_s
+    def url_for(base_path:, **params)
+      uri = URI.join(website_root, base_path)
+      query = Hash[URI.decode_www_form(uri.query.to_s)]
```

Comment:
what is the reason for having `Hash[URI.decode_www_form(uri.query.to_s)]` rather than just `{params}`?

Reply:
So there are two sets of params at play here: any existing params in the "base_path" (e.g. Brexit Checker criteria) and the ones we want to append. Because we don't know if the "base_path" has query params, we need to fish them out and merge them with the ones we want to add. `URI.decode_www_form` returns a key/value array, so we need to use `Hash` to get the initial hash to merge `params` with.

Does that help?

Reply:
ah that makes sense, got it thank you